lib/fluent/plugin/in_watch_process.rb

```diff
@@ -44,53 +54,175 @@ def shutdown
+        pid:integer
+        cpu_second:float
+        working_set:integer
+        virtual_memory_size:integer
+        handles:integer
+      ).join(",")
+
+      attr_reader :keys
+      attr_reader :command
+
+      def initialize(keys, command)
+        @keys = keys || DEFAULT_PARAMS.keys
+        @command = command || default_command
+      end
+
+      def default_types
+        DEFAULT_TYPES
       end
 
-      def OS.mac?
-       (/darwin/ =~ RUBY_PLATFORM) != nil
+      def parse_line(line)
+        values = line.chomp.strip.parse_csv.map { |e| e ? e : "" }
+        data = Hash[@keys.zip(values)]
+
+        unless data["start_time"].nil?
+          data["start_time"] = format_datetime(data["start_time"])
+        end
+
+        data
+      end
+
+      def default_command
+        command = [
+          command_ps,
+          pipe_filtering_normal_ps,
+          pipe_select_columns,
+          pipe_fixing_locale,
+          pipe_formatting_output,
+        ].join
+        "powershell -command \"#{command}\""
+      end
+
+      def command_ps
+        if @keys.include?("user")
+          # The "IncludeUserName" option is needed to get the username, but this option requires Administrator privilege.
+          "Get-Process -IncludeUserName"
+        else
+          "Get-Process"
+        end
+      end
+
+      private
+
+      def format_datetime(datetime)
+        Time.parse(datetime).to_s
+      end
+
+      def pipe_filtering_normal_ps
+        # There are some special processes that don't have some properties, such as the "Idle" process.
+        # Normally, these are specific to the system and are not important, so exclude them.
+        # Note: The same situation can occur in some processes if there are no Administrator privilege.
+        " | ?{$_.StartTime -ne $NULL -and $_.CPU -ne $NULL}"
+      end
+
+      def pipe_select_columns
+        columns = @keys.map { |key|
+          next unless DEFAULT_PARAMS.keys.include?(key)
+          DEFAULT_PARAMS[key]
+        }.compact
+
+        if columns.nil? || columns.empty?
+          raise "The 'keys' parameter is not specified correctly. [keys: #{@keys}]"
+        end
+
+        " | Select-Object -Property #{columns.join(',')}"
       end
 
-      def OS.unix?
-        !OS.windows?
+      def pipe_fixing_locale(format: "ddd MMM dd HH:mm:ss yyyy", locale: "en-US")
+        # In Windows, setting the "$env:Lang" environment variable is not effective in changing the format of the output.
+        # You can use "Datetime.ToString" method to change format of datetime values in for-each pipe.
+        # Note: About "DateTime.ToString" method: https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring
+        # Note: About "Custom date and time format strings": https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings
+        return "" unless @keys.include?("start_time")
+
+        " | %{
+          $_.StartTime = $_.StartTime.ToString(
```

Comment:
FWIW you can just do `StartTime.ToString("o")` here. You'll
get the following, regardless of what is the current locale.

```
2008-06-15T21:15:07.0000000
```

This is as per [the documentation](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring?view=net-5.0#System_DateTime_ToString_System_String_)


![Screenshot_2021-05-17_21-57-14](https://user-images.githubusercontent.com/8974561/118492349-e4e3b500-b75a-11eb-8192-da2b726cca04.png)